lib/spack/env/cc

```diff
@@ -58,59 +62,105 @@ parameters=(
 # Test command is used to unit test the compiler script.
 #   SPACK_TEST_COMMAND
 
-# die()
+# die MESSAGE
 # Prints a message and exits with error 1.
-function die {
+die() {
     echo "$@"
     exit 1
 }
 
-# read input parameters into proper bash arrays.
-# SYSTEM_DIRS is delimited by :
-IFS=':' read -ra SPACK_SYSTEM_DIRS <<< "${SPACK_SYSTEM_DIRS}"
+# empty VARNAME
+# whether a variable is empty
+empty() {
+    eval "test -z \"\${$1}\""
+}
+
+# contains VARNAME SUBSTRING
+# whether a varaible's value contains a substring
+contains() {
+    eval "test \"\${$1#*$2}\" != \"\${$1}\""
+}
+
+# dirlist VARNAME
+# Is VARNAME a list of directories? It is if it ends with _dirs.
+dirlist() {
+    test "${1%%_dirs}" != "${1}" -o "${1%%_DIRS}" != "${1}"
+}
 
-# SPACK_<LANG>FLAGS and SPACK_LDLIBS are split by ' '
-IFS=' ' read -ra SPACK_FFLAGS   <<< "$SPACK_FFLAGS"
-IFS=' ' read -ra SPACK_CPPFLAGS <<< "$SPACK_CPPFLAGS"
-IFS=' ' read -ra SPACK_CFLAGS   <<< "$SPACK_CFLAGS"
-IFS=' ' read -ra SPACK_CXXFLAGS <<< "$SPACK_CXXFLAGS"
-IFS=' ' read -ra SPACK_LDFLAGS  <<< "$SPACK_LDFLAGS"
-IFS=' ' read -ra SPACK_LDLIBS   <<< "$SPACK_LDLIBS"
+# append LISTNAME ELEMENT [SEP]
+#
+# Appends element to list, assuming list is separted by SEP.
+# Handles empty lists and single-element lists.
+append() {
+    varname="$1"
+    elt="$2"
+
+    # if sep is provided, use it. If not, sep is:
+    # ':' if the varname ends in _dirs
+    # ' ' for other var names
+    if dirlist "$varname"; then
+        sep="${3:-:}"
+    else
+        sep="${3:- }"
+    fi
```

Comment:
Thoughts on using a `Tab`-delimited rather than space-delimited string for the pseudo-array? A quick test showed it fixed string splitting for things like `-L/other/lib dir`, and it has the nice property that it's still split naturally by the default IFS if that's desirable.  

```suggestion
    # '<Tab>' for other var names
    if dirlist "$varname"; then
        sep="${3:-:}"
    else
        sep="${3:-	}"
    fi
```

Reply:
used bell (^G) instead